### DISCUSSION TITLE
Running collection requests in pre request / post response requests

### DISCUSSION THREAD

**Discussion by scflode** [usebruno/bruno#535] [Ideas]
Just was wondering about how to run requests (e.g. authentication) from within a pre request script. It's all possible with Axios manually and you can even use variables defined.

The point of the idea here is if it's possible to have something like `bru.run("Request name")` basically wrapping the Axios ceremony.

**Comment by helloanoop** [comment]
Yes, We will support this soon.

There is an alternative too.
You can basically define the entire axios ceremony inside a file.
And then require that file and just call it - see https://github.com/usebruno/bruno/discussions/385#discussioncomment-7224068 

If you need something to be run for every request in the collection, Bruno now has support for collection level scripting

**Reply by scflode** [reply]
Oh thanks. Thought about the script part as well. Went with a collection script for now. It’s perfectly fine 👍 for what I’m using. 
Bruno is already extremely powerful 💪

**Comment by igel1** [comment]
> Yes, We will support this soon.

Is there a spot, where I can track the progress of implementation of this feature?
I am also highly interested in such a "bru.run(request name)" feature.

**Reply by groblochon** [reply]
This one works fine : `bru.setNextRequest("request name");`

**Comment by alexkunde** [comment]
We would also be highly interested in the feature to run one request and be able to use the outputs. It would improve our testing codebase by a lot.

**Comment by tcasada** [comment]
> This one works fine : `bru.setNextRequest("request name");`

That seems to successfully run the request _after_ the current request when I put it in the Pre Request Script section.  First run with this syntax fails due to missing token, but a quick subsequent run pulls that "nextRequest" from a stored variable perfectly.  I think the feature we are looking for is run this "pre request" as the original poster said, bru.run or maybe bru.runRequest("request name").

Thanks for sharing that idea though @groblochon!

**Reply by thiagomini** [reply]
Is this solved?

**Reply by jkblume** [reply]
I think is is already solved: https://docs.usebruno.com/testing/script/javascript-reference#runrequest
